core/core-configuration-layer.el

```diff
@@ -1114,12 +1172,15 @@ Returns nil if the directory is not a category."
       (when (string-match "^+" dirname)
         (intern (substring dirname 1))))))
 
-(defun configuration-layer/discover-layers ()
-  "Initialize `configuration-layer--indexed-layers' with layer directories."
+(defun configuration-layer/discover-layers (&optional no-new-table)
+  "Initialize `configuration-layer--indexed-layers' with layer directories.
+If `no-new-table' is true, the layer index is not cleared before
```

Comment:
prefer:

> If NO-NEW-TABLE is non-nil ...

We should invert the purpose of this parameter, by default we don't create new structure but if `CREATE-NEW-TABLE` is non nil then we do.